Instruments/source/CacheTupleProducer.cxx

```diff
@@ -0,0 +1,161 @@
+        auto outputFile = root_ext::CreateRootFile(args.output_file());
+        auto originalTuple = ntuple::CreateEventTuple(args.tree_name(),originalFile.get(),true,ntuple::TreeState::Full);
+        CacheTuple cache(args.tree_name(), outputFile.get(), false);
+        auto summaryTuple = ntuple::CreateSummaryTuple("summary", originalFile.get(), true, ntuple::TreeState::Full);
+        summaryTuple->GetEntry(0);
+        SummaryInfo summaryInfo(summaryTuple->data(), Parse<Channel>(args.tree_name()), args.trigger_cfg());
+        for(const auto& event : *originalTuple) {
+            FillCacheTuple(cache, event,summaryInfo);
+            cache.Fill();
+        }
+
+        cache.Write();
+
+    }
+
+private:
+
+    void FillCacheTuple(CacheTuple& cacheTuple, const ntuple::Event& event,const SummaryInfo& summaryInfo) const
+    {
+        static const std::map<Channel, std::vector<std::string>> triggerPaths = {
+            { Channel::ETau, { "HLT_Ele32_WPTight_Gsf_v", "HLT_Ele35_WPTight_Gsf_v", "HLT_Ele24_eta2p1_WPTight_Gsf_LooseChargedIsoPFTau30_eta2p1_CrossL1_v" } },
+            { Channel::MuTau, { "HLT_IsoMu24_v", "HLT_IsoMu27_v", "HLT_IsoMu20_eta2p1_LooseChargedIsoPFTau27_eta2p1_CrossL1_v" } },
+            { Channel::TauTau, { "HLT_DoubleTightChargedIsoPFTau35_Trk1_TightID_eta2p1_Reg_v",
+                "HLT_DoubleMediumChargedIsoPFTau40_Trk1_TightID_eta2p1_Reg_v",
+                "HLT_DoubleTightChargedIsoPFTau40_Trk1_eta2p1_Reg_v"} },
+            { Channel::MuMu, { "HLT_IsoMu24_v", "HLT_IsoMu27_v" } },
+        };
+        const Channel channel = Parse<Channel>(args.tree_name());
+
+        if(event.extraelec_veto || event.extramuon_veto) return;
+
+        //JetOrdering jet_ordering = run_period == Period::Run2017 ? JetOrdering::DeepCSV : JetOrdering::CSV;
+        JetOrdering jet_ordering = JetOrdering::DeepFlavour;
+        std::set<size_t> Htt_indexes;
+        std::set<std::pair<size_t,size_t>> HH_indexes;
+        for(unsigned n = 0; n < signalObjectSelectors.size(); ++n){
+            for(unsigned h = 0; h < unc_sources.size(); ++h){
+                for(int l = -1; l < 2; ++l){
+                    cacheTuple().run = event.run;
+                    cacheTuple().lumi = event.lumi;
+                    cacheTuple().evt = event.evt;
+                    SignalObjectSelector signalObjectSelector = signalObjectSelectors.at(n);
+                    boost::optional<EventInfoBase> event_info_base = CreateEventInfo(event,signalObjectSelector,&summaryInfo,run_period,jet_ordering);
+                    if(!event_info_base.is_initialized()) continue;
+                    if(!event_info_base->GetTriggerResults().AnyAcceptAndMatchEx(triggerPaths.at(channel), event_info_base->GetFirstLeg().GetMomentum().pt(),
+                                                                                                        event_info_base->GetSecondLeg().GetMomentum().pt())) continue;
+                    if(!event_info_base->HasBjetPair()) continue;
```

Comment:
In the future, we may want to have svfit for 0j and 1j control regions. Add a command-line argument to control whatever this cut should be applied.